packages/orchestration/src/joiner/helpers.ts

```diff
@@ -0,0 +1,59 @@
+import { RemoteJoinerQuery } from "@medusajs/types"
+
+export function toRemoteJoinerQuery(obj: any): RemoteJoinerQuery {
+  const remoteJoinerQuery: RemoteJoinerQuery = {
+    alias: "",
+    fields: [],
+    expands: [],
+  }
+
+  let isEntryPoint = true
+  function extractRecursive(obj, parentName = "") {
+    for (let key in obj) {
+      const value = obj[key]
+
+      const canExpand =
+        typeof value === "object" &&
+        !["fields", "__args", "__directives"].includes(key)
+
+      if (canExpand) {
+        const entityName = parentName ? `${parentName}.${key}` : key
+        const expandObj: any = {
+          property: entityName,
+        }
+
+        const reference = isEntryPoint ? remoteJoinerQuery : expandObj
+
+        if (value.__args) {
+          reference.args = Object.entries(value.__args).map(
+            ([name, value]) => ({
+              name,
+              value,
+            })
+          )
+        }
+
+        if (value.__directives) {
+          reference.directives = Object.entries(value.__args).map(
+            ([name, value]) => ({ name, value })
+          )
+        }
+
+        reference.fields = value.fields
+
+        if (isEntryPoint) {
+          remoteJoinerQuery.alias = key
+          isEntryPoint = false
+        } else {
+          remoteJoinerQuery.expands!.push(expandObj)
+        }
+
+        extractRecursive(value, entityName)
```

Comment:
comment: based on previous comment if accepted, pass isEntryPoint as the third arg